src/reducers/tags/__tests__/reducer.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { tagsReducer } from "../reducer"
+import { emptyAction } from "../../actions"
+import { setTagsAction, saveTagAction, deleteTagAction } from "../actions"
+import { Map } from "immutable"
+
+describe("tagsReducer", () => {
+  it("returns initials state", () => {
+    const state = tagsReducer(Map(), emptyAction())
+
+    expect(state).toEqual(Map())
+  })
+
+  it("sets tags", () => {
+    const tags = Map([["fake_id", { id: "fake_id", color: "black&white" }]])
+    const action = setTagsAction(tags)
+    const state = tagsReducer(Map(), action)
+
+    expect(state).toEqual(tags)
+  })
+
+  it("adds tag", () => {
+    const tag = { id: "fake_id", color: "black&white" }
+    const action = saveTagAction(tag)
+    const state = tagsReducer(Map(), action)
+
+    expect(state).toEqual(Map([[tag.id, tag]]))
+  })
+
+  it("delete tag", () => {
+    const tag = { id: "fake_id", color: "black&white" }
+    const action = deleteTagAction(tag)
+    const state = tagsReducer(Map(), action)
```

Comment:
Initial state should be `Map([[tag.id, tag]])`
If initial state is empty it will not test delete action since there is nothing to delete.